fractal_server/app/schemas/manifest.py

```diff
@@ -0,0 +1,41 @@
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import TypeVar
+
+from pydantic import BaseModel
+from pydantic import validator
+
+
+__all__ = ("TaskManifestV1", "ManifestV1")
+
+
+class _TaskManifestBase(BaseModel):
+    name: str
+    executable: Path
+    input_type: str
+    output_type: str
+    default_args: Dict[str, Any]
+    meta: Dict[str, Any]
+
+
+TaskManifestType = TypeVar("TaskManifestType", bound=_TaskManifestBase)
+
+
+class _ManifestBase(BaseModel):
+    manifest_version: str
+    task_list: List[TaskManifestType]  # type: ignore
+
+
+class TaskManifestV1(_TaskManifestBase):
+    pass
+
+
+class ManifestV1(_ManifestBase):
+    task_list: List[TaskManifestV1]
+
+    @validator("manifest_version")
```

Comment:
Another side-comment (not needed for this PR)
It would be useful to expose the validator as a simple command-line tool, through a simple wrapper, for the user to validate their custom manifest (for custom tasks). Something like
```
validate_manifest.py /path/to/my/manifest
```

Reply:
> Another side-comment (not needed for this PR) It would be useful to expose the validator as a simple command-line tool, through a simple wrapper, for the user to validate their custom manifest (for custom tasks). 

Indeed, that's why I decided to make a pydantic schema for it. Keep in mind that when the refactoring will be over we could either mirror or move `app/schemas` in `fractal_common`. At that point it is trivial to add the validation utility to the client cli.